src/extensions/fixed-columns/bootstrap-table-fixed-columns.js

```diff
@@ -56,58 +51,244 @@ $.BootstrapTable = class extends $.BootstrapTable {
     }
 
     this.initFixedColumnsBody()
+    this.initFixedColumnsEvents()
+  }
+
+  trigger (...args) {
+    super.trigger(...args)
+
+    if (!this.fixedColumnsSupported()) {
+      return
+    }
 
-    this.$fixedBody.css({
-      top: 0,
-      width: this.getFixedColumnsWidth(),
-      height: this.$tableHeader.outerHeight(true) + this.$tableBody.outerHeight(true)
+    if (args[0] === 'post-header') {
+      this.initFixedColumnsHeader()
+    } else if (args[0] === 'scroll-body') {
+      if (this.needFixedColumns && this.options.fixedNumber) {
+        this.$fixedBody.scrollTop(this.$tableBody.scrollTop())
+      }
+
+      if (this.needFixedColumns && this.options.fixedRightNumber) {
+        this.$fixedBodyRight.scrollTop(this.$tableBody.scrollTop())
+      }
+    }
+  }
+
+  updateSelected () {
+    super.updateSelected()
+
+    if (!this.fixedColumnsSupported()) {
+      return
+    }
+
+    this.$tableBody.find('tr').each((i, el) => {
+      const index = $(el).data('index')
+      const classes = $(el).attr('class')
+
+      const inputSelector = `[name="${this.options.selectItemName}"]`
+      const $input = $(el).find(inputSelector)
+
+      if (typeof index === undefined) {
+        return
+      }
+
+      if (this.$fixedBody && this.options.fixedNumber) {
+        const $tr = this.$fixedBody.find(`tr[data-index="${index}"]`)
+        $tr.attr('class', classes)
+
+        if ($input.length) {
+          $tr.find(inputSelector).prop('checked', $input.prop('checked'))
+        }
+
+        if (this.$selectAll.length) {
+          this.$fixedHeader.add(this.$fixedBody)
+            .find('[name="btSelectAll"]')
+            .prop('checked', this.$selectAll.prop('checked'))
+        }
+      }
+
+      if (this.$fixedBodyRight && this.options.fixedRightNumber) {
+        const $tr = this.$fixedBodyRight.find(`tr[data-index="${index}"]`)
+        $tr.attr('class', classes)
+
+        if ($input.length) {
+          $tr.find(inputSelector).prop('checked', $input.prop('checked'))
+        }
+
+        if (this.$selectAll.length) {
+          this.$fixedHeaderRight.add(this.$fixedBodyRight)
+            .find('[name="btSelectAll"]')
+            .prop('checked', this.$selectAll.prop('checked'))
+        }
+      }
     })
+  }
 
+  initFixedColumnsHeader () {
+    if (this.options.height) {
+      this.needFixedColumns = this.$tableHeader.width() < this.$tableHeader.find('table').width()
+    } else {
+      this.needFixedColumns = this.$tableBody.width() < this.$tableBody.find('table').width()
+    }
+
+    if (this.needFixedColumns && this.options.fixedNumber) {
+      this.$fixedColumns.find('.fixed-table-header').remove()
```

Comment:
Why removing .fixed-table-header and then find it again?

Reply:
The header will be recreated, thats the reason i guess.

Reply:
This 3 lines will be used often, maybe create a function for that ?